src/extensions/anvil/algorithm.ts

```diff
@@ -262,13 +490,17 @@ function qValue(currentStats: HphobHphil, initialStats: HphobHphil): number {
 }
 
 export function isInMembranePlane(testPoint: Vec3, normalVector: Vec3, planePoint1: Vec3, planePoint2: Vec3): boolean {
-    const d1 = -Vec3.dot(normalVector, planePoint1);
-    const d2 = -Vec3.dot(normalVector, planePoint2);
-    const d = -Vec3.dot(normalVector, testPoint);
-    return d > Math.min(d1, d2) && d < Math.max(d1, d2);
+    const d1 = -v3dot(normalVector, planePoint1);
+    const d2 = -v3dot(normalVector, planePoint2);
+    return _isInMembranePlane(testPoint, normalVector, Math.min(d1, d2), Math.max(d1, d2));
+}
+
+function _isInMembranePlane(testPoint: Vec3, normalVector: Vec3, min: number, max: number): boolean {
+    const d = -v3dot(normalVector, testPoint);
+    return d > min && d < max;
 }
 
-// generates a defined number of points on a sphere with radius = extent around the specified centroid
+/** Generates a defined number of points on a sphere with radius = extent around the specified centroid */
 function generateSpherePoints(ctx: ANVILContext, numberOfSpherePoints: number): Vec3[] {
```

Comment:
Does this take symmetry into account? Would a vector from south to north be the same for testing as one from north to south?

Reply:
Somewhat. The direction is symmetric but the mirrored sphere point usually leads to slightly different results if the extent of the structure is not exactly divisible by the stepSize. Every direction is evaluated twice with a slight offset, but no efforts are made to choose a good offset (like half the stepSize).

Given that, one hemisphere could be sampled by double the number of sphere points while the other hemisphere gets ignored.

Reply:
fcac1a6 moves all sphere points to one hemisphere. I've ran some test cases and the predictions are fine.

Reply:
Very nice, thank you